docs/source/trainer/numerics.md

```diff
@@ -0,0 +1,73 @@
+- `fp16`: [Half-precision Floating-point](https://en.wikipedia.org/wiki/Half-precision_floating-point_format)
+- `bf16`: [Brain Floating-point](https://en.wikipedia.org/wiki/Bfloat16_floating-point_format)
+
+### Mixed Precision
+
+Using half-precision number formats can boost model throughput, though not without issue.  The reduced representable range and sensitivity of these formats can cause several problems during training such as:
+
+- Gradient underflow: Very small values can get zeroed out
+- Incorrect weight updates: Small gradient values can lead to no weight updates
+- Activation/Loss overflow: Very large values can roll over to representations of [Infinity](https://en.wikipedia.org/wiki/Infinity#Computing)(`Inf`)or [Not a Number](https://en.wikipedia.org/wiki/NaN) (`NaN`)
+
+The result is reduced model accuracy.
+
+The solution is to perform mixed precision training, where both single and half-precision formats are utilized strategically.  Mixed precision training is performed as follows:
+
+1. Compute the forward pass and loss in half-precision, *except* for computations that can cause activations/loss overflow, which are performed in single-precision (e.g., BatchNorm)
+2. Compute the backpropagation pass in half-precision
+3. To increase numerical sensitivity, convert the model to single-precision and perform the optimizer step, enabling the weight update to be done more precisely
+4. Convert the model back to half-precision
+
+The procedure above mitigates the issues with overflow and imprecise weight updates. 
+
+#### Gradient Scaling
+
+In order to prevent gradient underflow, mixed precision training uses a technique called **Gradient Scaling** where the loss is multiplied by a scaling factor *before* backpropagation is performed (step 2 above). This trick scales the computed gradient values to a range where they cannot underflow.  Once the model has been converted to single-precision following the backward pass, the gradients are divided by the scaling factor to restore the gradient values to the original range before stepping the optimizer.
+
+### AMP
+
+Composer supports automatic mixed precision (AMP) training using PyTorch’s {mod}`torch.cuda.amp` package. The Composer {class}`~.Trainer` performs all the heavy lifting and tensor conversions automatically; the user simply has to set `precision='amp'` when initializing the {class}`~.Trainer`.
+
+Another advantage of using Composer is support for optimizer closures when using mixed precision.  A closure is responsible for clearing the gradients and re-computing the loss, as required by some optimizers. The {class}`~.Trainer` supports closures when using `amp` if the optimizer object has the attribute {attr}`~.SAMOptimizer._step_supports_amp_closure` set to `True`.  Please see the [SAM Method Card](../method_cards/sam.md) for an implementation example on how closures are used.
```

Comment:
Do we have an API docs reference for `step_supports_amp_closure`? If not, we should somewhere. But for now, could you rephrase this as an example of how SAM is an example of how to use this flag?

Reply:
We don't, I added a cross-reference to the `SAMOptimizer` API Reference here for the time being.